core/global/platform.gd

```diff
@@ -374,94 +385,303 @@ func _read_cpu_info() -> CPUInfo:
+	for card in found_cards:
+		for info in vulkan_info:
+			if card.vendor_id == info[0] and card.device_id == info[1]:
+				card.device_type = info[2]
+				logger.debug("Assigning type " + card.device_type + " to device " + card.device)
+
+	return found_cards
+
+
+## Helper function that simplifies reading id values from a given path.
+func _get_card_property_from_path(path: String) -> String:
+	return FileAccess.get_file_as_string(path).lstrip("0x").to_lower().strip_escapes()
+
+
+## Returns a PackedStringArray that includes the Vendor Name, Device Name,
+## and Subdevice Name as defined in /usr/share/hwdata/pci.ids byt matching
+## the id values derived from /sys/class/drm/cardX/device/<property> from
+## the list <vendor/device/subsystem_vendor/subsystem_device>.
+func get_device_from_card(cardinfo: CardInfo) -> PackedStringArray:
+	var hwids := FileAccess.open(pci_ids_path, FileAccess.READ)
+	var card_vendor_name: String
+	var vendor_found: bool = false
+	var card_device_name: String
+	var device_found: bool = false
+	var card_subdevice_name: String
+	logger.debug("Getting device info from: " + cardinfo.vendor_id + " " + cardinfo.device_id + " " + cardinfo.subvendor_id + " " + cardinfo.subdevice_id)
+	while not hwids.eof_reached():
+		var line := hwids.get_line()
+		var line_clean := line.strip_escapes()
+
+		if line.begins_with("\t") and not vendor_found:
+			continue
+		if line.begins_with(cardinfo.vendor_id):
+			card_vendor_name = line.lstrip(cardinfo.vendor_id).strip_edges()
+			logger.debug("Found vendor_name: " + card_vendor_name)
+			vendor_found = true
+			continue
+		if vendor_found and not line.begins_with("\t"):
+			if line.begins_with("#"):
+				continue
+			logger.debug("Got to end of vendor list. Device not found")
+			break
 
-# Run glxinfo and return the data from it.
-# TODO: Maybe use vulkaninfo? Need a way to get vendor string in that. It can
-# output to JSON so it might be easier to get more info like driver name and info,
-# device type (dedicated or integrated), etc.
-func _get_glxinfo() -> Array:
+		if line.begins_with("\t\t") and not device_found:
+			continue
+
+		if line_clean.begins_with(cardinfo.device_id):
+			card_device_name = line_clean.lstrip(cardinfo.device_id).strip_edges()
+			logger.debug("Found device_name: " + card_device_name)
+			device_found = true
+
+		if device_found and not line.begins_with("\t\t"):
+			logger.debug("Got to end of device list. Subdevice not found")
+			break
+
+		var prefix := cardinfo.subvendor_id + " " + cardinfo.subdevice_id
+		if line_clean.begins_with(prefix):
+			card_subdevice_name = line.lstrip(prefix)
+			logger.debug("Found subdevice_name: " + card_subdevice_name)
+			break
+
+	return [card_vendor_name, card_device_name, card_subdevice_name]
+
+
+func _get_cards_from_vulkan() ->Array[PackedStringArray]:
+	var vulkan_cards: Array[PackedStringArray] = []
+	var args = ["-c", "vulkaninfo", "--summary"]
+	var output: Array = _do_exec("bash", args)
```

Comment:
I tried this initially. The output was getting truncated for some reason. Seems to be working now so I'll revert.

Reply:
If you do ever need to use bash, it should be in the form of: 

`_do_exec("bash", ["-c", "vulkaninfo --summary"])`

instead of

`_do_exec("bash", ["-c", "vulkaninfo", "--summary"])`